sdk/cosmos/src/operations/create_reference_attachment.rs

```diff
@@ -0,0 +1,142 @@
+use crate::headers::from_headers::*;
+use crate::prelude::*;
+use crate::resources::Attachment;
+use crate::ResourceQuota;
+use azure_core::headers::{etag_from_headers, session_token_from_headers};
+use azure_core::prelude::*;
+use azure_core::SessionToken;
+use azure_core::{collect_pinned_stream, Request as HttpRequest, Response as HttpResponse};
+use chrono::{DateTime, Utc};
+
+#[derive(Debug, Clone)]
+pub struct CreateReferenceAttachmentOptions<'a, 'b> {
+    attachment_client: &'a AttachmentClient,
+    activity_id: Option<ActivityId<'b>>,
+    consistency_level: Option<ConsistencyLevel>,
+}
+
+impl<'a, 'b> CreateReferenceAttachmentOptions<'a, 'b> {
+    pub fn new(attachment_client: &'a AttachmentClient) -> Self {
+        Self {
+            attachment_client,
+            activity_id: None,
+            consistency_level: None,
+        }
+    }
+}
+
+impl<'a, 'b> CreateReferenceAttachmentOptions<'a, 'b> {
+    setters! {
+        activity_id: &'b str => Some(ActivityId::new(activity_id)),
+        consistency_level: ConsistencyLevel => Some(consistency_level),
+    }
+}
+
+impl<'a, 'b> CreateReferenceAttachmentOptions<'a, 'b> {
+    pub(crate) fn decorate_request<'c, M, C>(
+        &self,
+        request: &mut HttpRequest,
+        media: M,
+        content_type: C,
+    ) -> Result<(), crate::Error>
+    where
+        M: AsRef<str>,
+        C: Into<ContentType<'c>>,
+    {
+        azure_core::headers::add_optional_header2(&self.activity_id, request)?;
+        azure_core::headers::add_optional_header2(&self.consistency_level, request)?;
+
+        crate::cosmos_entity::add_as_partition_key_header_serialized2(
+            self.attachment_client
+                .document_client()
+                .partition_key_serialized(),
+            request,
+        );
+
+        #[derive(Debug, Serialize)]
+        struct Request<'r> {
+            pub id: &'r str,
+            #[serde(rename = "contentType")]
+            pub content_type: &'r str,
+            pub media: &'r str,
+        }
+
+        let body = azure_core::to_json(&Request {
+            id: self.attachment_client.attachment_name(),
```

Comment:
We shouldn't keep a reference to the attachment_client in this options object. Instead we should take the serialized partition key and attachment name as arguments to this function.

Reply:
I agree with this, but worried that this refactoring was overreach for an MR that was only pipeline-ifying.  However if you're happy for this all to go in one, can take a look later.